src/frontend-samples/snow-decoration-sample/SnowDecorationWidget.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect } from "react";
+import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, UiItemsProvider, WidgetState } from "@bentley/ui-abstract";
+import { Button, Select, Slider, Toggle } from "@bentley/ui-core";
+import SnowDecorationApi from "./SnowDecorationApi";
+import { SnowParams } from "./SnowDecorator";
+import { IModelApp } from "@bentley/imodeljs-frontend";
+import "./SnowDecoration.scss";
+
+const windRange = 600;
+let lock: Promise<void> | false = false;
+
+const SnowDecorationWidget: React.FunctionComponent = () => {
+  const [propsName, setPropsName] = React.useState<string>(SnowDecorationApi.predefinedProps.keys().next().value);
+  const [wind, setWind] = React.useState<number>(0);
+  const [particleDensity, setParticleDensity] = React.useState<number>(0);
+  const [pauseEffect, setPauseEffect] = React.useState<boolean>(false);
+
+  useEffect(() => {
+    // useEffect can not be async, so the "lock" variable will prevent the creating another decorator before the first is completed.
+    const vp = IModelApp.viewManager.selectedView;
+    if (!vp || lock)
+      return;
+    lock = (async () => {
+      const props = SnowDecorationApi.predefinedProps.get(propsName)!;
+      SnowDecorationApi.createSnowDecorator(vp, props).then(() => {
+        setParticleDensity(props.params.particleDensity);
+        setWind(props.params.windVelocity);
+        setPauseEffect(false);
+        lock = false;
+      });
+    })();
+  }, [propsName]);
+
+  useEffect(() => {
+    configureEffect({ particleDensity });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [particleDensity]);
+
+  useEffect(() => {
+    configureEffect({ windVelocity: wind });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [wind]);
+
+  useEffect(() => {
+    configureEffect({ pauseEffect });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pauseEffect]);
+
+  /** Configures active snow decorators (should only ever be one in this sample). */
+  const configureEffect = (params: Partial<SnowParams> & { pauseEffect?: boolean }) => {
```

Comment:
Effect not applied when switching between models

Reply:
Fixed.